File: 2298-count-integers-with-even-digit-sum/2298-count-integers-with-even-digit-sum.cpp

class Solution {

public:

    int countEven(int num) {

        int cnt = 0;

        if (num >= 2) cnt = 1;

        for (int i = 4; i <= num; i++) {

            int temp = 0;

            temp += i % 10;

            temp += (i / 10);

            temp += (i / 100);

            temp += (i / 1000);

            if (temp % 2 == 0) cnt++;            

        }

        return cnt;

    }

};

File: 2298-count-integers-with-even-digit-sum/2298-count-integers-with-even-digit-sum.java
class Solution {
    public int digitSum(int num) {
        int sum = 0;
        while(num > 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }
    public int countEven(int num) {
        int ans = 0;
        for (int i = 2; i <= num; i++) {
            if(digitSum(i) % 2 == 0) {
                ans++;
            }
        }
        return ans;
    }
}